langchain/llms/gooseai.py

```diff
@@ -0,0 +1,134 @@
+"""Wrapper around GooseAI API."""
+import logging
+
+from typing import Any, Dict, List, Mapping, Optional
+
+from pydantic import BaseModel, Extra, Field, root_validator
+
+from langchain.utils import get_from_dict_or_env
+from langchain.llms.base import LLM
+
+logger = logging.getLogger(__name__)
+
+class GooseAI(LLM, BaseModel):
```

Comment:
If a lot of this is very similar to the OpenAI class, should we abstract that similarity and then create this class from that abstraction?

Reply:
I was thinking about just adding an abstraction similar to AzureOpenAI but I was concerned that if GooseAI or OpenAI decide to change their APIs in some sort of manner it could cause conflicts. I did not know if it would be easier to diagnose or maintain if they are separated. 

I am completely open to just adding a similar class GooseAI class abstraction if you think that is better. Whatever works best!